packages/flutter_tools/lib/src/reporting/events.dart

```diff
@@ -249,3 +249,40 @@ class CodeSizeEvent extends UsageEvent {
 class ErrorHandlingEvent extends UsageEvent {
   ErrorHandlingEvent(String parameter) : super('error-handling', parameter, flutterUsage: globals.flutterUsage);
 }
+
+/// The category for analytics events related to null safety.
+const String kNullSafetyCategory = 'null-safety';
+
+/// Emit various null safety analytic events.
+///
+/// 1. The current null safety runtime mode.
+/// 2. The % of dependencies that are not migrated.
+/// 3. The main language version.
+void collectLanguageVersionEvents(
```

Comment:
I'd suggest following suit and inheriting from `UsageEvent`. You can override the `send()` method to actually send several events.